src/record/record.js

```diff
@@ -151,17 +151,18 @@ Record.prototype.set = function( pathOrData, dataOrCallback, callback ) {
 	var newValue = jsonPath.set( oldValue, path, data, this._options.recordDeepCopy );
 
 	if ( oldValue === newValue ) {
+		if ( typeof callback === 'function' ) utils.setImmediate( callback, null );
```

Comment:
I don't really understand this comment, can you elaborate? :)

As I see it, it does not result in a new write when the value is the same, since it does a return on next line. The issue was however that it never called the callback in this case.

Resulting in the scenario ```record.set(something, something, function doStuff)``` where _doStuff_ was not always called.